### PR TITLE
🌱 logging: add removal note, cleanup no-op code

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related: https://github.com/kubernetes-sigs/cluster-api/issues/6069

### CODE REVIEW

**Review by killianmuldoon — APPROVED** [review]
/lgtm

File: docs/book/src/developer/providers/v1.3-to-v1.4.md [hunk 1]
```diff
@@ -41,6 +41,11 @@ maintainers of providers and consumers of our Go API.
 - `clusterctl upgrade apply` no longer requires a namespace when updating providers. It is now optional and in a future release it will be deprecated. The new syntax is `[namespace/]provider:version`.
 - `WatchDeploymentLogs` is changed to `WatchDeploymentLogsByName`, it works same as before. Another function `WatchDeploymentLogsByLabelSelector` is added to stream logs of deployment by label selector.
 - Cluster API controllers are now using an explicit security context by default.
+- It is recommended to drop usages of `logs.AddFlags(fs, logs.SkipLoggingConfigurationFlags())`. It was previously used to configure deprecated logging flags, but with the bump to component-base
```
Comment (sbueringer) [inline docs/book/src/developer/providers/v1.3-to-v1.4.md:44]: @oscr I added this to the provider migration doc as it's relevant for provider authors as well.

But we also want to notify users (https://github.com/kubernetes-sigs/cluster-api/issues/6069).

What is the best way to do this?
Is the current plan still to add another user-facing doc somewhere here? (xref: https://github.com/kubernetes-sigs/cluster-api/issues/7672#issuecomment-1361785064)

**Comment by sbueringer** [comment]
cc @chrischdi Just fyi, given you recently updated the Runtime Extensions doc.

**Review by ykakarap — APPROVED** [review]
/lgtm

**Comment by sbueringer** [comment]
/approve

given lgtm's above.

@ykakarap Can we also close https://github.com/kubernetes-sigs/cluster-api/issues/6069 and carry over https://github.com/kubernetes-sigs/cluster-api/pull/7955#discussion_r1081689610 into https://github.com/kubernetes-sigs/cluster-api/issues/7672?

This would allow us to close the 1.26 issue.

**Comment by ykakarap** [comment]
> /approve
> 
> given lgtm's above.
> 
> @ykakarap Can we also close #6069 and carry over [#7955 (comment)](https://github.com/kubernetes-sigs/cluster-api/pull/7955#discussion_r1081689610) into #7672?
> 
> This would allow us to close the 1.26 issue.

+1. Added an item in the release tracking issue for the Communications manager to add a user facing doc and also referenced this issue under that item.

File: docs/book/src/developer/providers/v1.3-to-v1.4.md (hunk 1 above)
Comment (oscr) [reply docs/book/src/developer/providers/v1.3-to-v1.4.md:44]: @sbueringer Yes, as suggested we have two release documents: one for [users](https://hackmd.io/Ou7CmslpQUaiwXVPuHE-BA) and one for [implementors](https://hackmd.io/LB5o_LHHTdOMb6IAWswljg).

Comment (sbueringer) [reply docs/book/src/developer/providers/v1.3-to-v1.4.md:44]: The idea with having those documents in the repo was that everyone who implements a change can add the corresponding notes there and it's all transparent

We have good experience with that witht he provider implementers doc, e.g. reviewers are regularly asking folks to add the notes in the PRs where they implement the change.

I have some doubts if we can keep this up if we have 2 hackmd docs that are only ~ accessible to the docs team.

Comment (sbueringer) [reply docs/book/src/developer/providers/v1.3-to-v1.4.md:44]: (Also this way provider adoption early CAPI versions from main can see what they have to adopt to and modify according to the latest changes to the providers doc, same for users)

Comment (oscr) [reply docs/book/src/developer/providers/v1.3-to-v1.4.md:44]: You are absolutely right Stefan. I didn't reflect sufficiently on the question asked. What you suggested is a much better solution.

Comment (sbueringer) [reply docs/book/src/developer/providers/v1.3-to-v1.4.md:44]: No worries, good that we have discussions like this 😀

**Pull request merged**